### DISCUSSION TITLE
Open XML SDK 2.5 Productivity Tool

### DISCUSSION THREAD

**Discussion by tomjebo** [OfficeDev/Open-XML-SDK#1008] [General]
The Open XML SDK 2.5 Productivity Tool was recently removed from the Microsoft Download Center because of concerns with SHA-1 signatures. There have been many requests for it in the meantime, so we’re now making it available again with a SHA-256 signature for the MSI container (while the files inside the MSI are still signed with SHA-1).
This MSI is provided **AS IS**, to address the needs of developers wanting to use the Productivity Tool to get a quick start on generating Open XML 2.5 source code. It has not been updated to the latest version of the SDK (2.13.1).

<hr /><em>This discussion was created from the release <a href='https://github.com/OfficeDev/Open-XML-SDK/releases/tag/'>Open XML SDK 2.5 Productivity Tool</a>.</em>

**Comment by tauheedul** [comment]
Hi, will this be updated in the future to use the current version of the SDK 2.17.1?
and support Validation of the newer versions of Excel documents e.g. 2016/2019/2021/365?

![image](https://user-images.githubusercontent.com/9335220/183070471-7796b2f5-2d94-404c-9b2a-295a787dd6ec.png)

At present the Productivity Tool can Reflect code (generate code) for a worksheet (to create an OpenXML document).

![image](https://user-images.githubusercontent.com/9335220/183071414-d6a29885-d9bc-4721-9d75-18eb0d1825f5.png)

Can this be extended to generate code to Read the selected worksheet?

**Comment by AdamShakhabov** [comment]
> _This discussion was created from the release [Open XML SDK 2.5 Productivity Tool](https://github.com/OfficeDev/Open-XML-SDK/releases/tag/)._

The link is broken or incorrect.

**Reply by tauheedul** [reply]
The link is referenced under the release 2.5
Open XML SDK 2.5 Productivity Tool: https://github.com/OfficeDev/Open-XML-SDK/releases/tag/v2.5

**Comment by mdkimzey943** [comment]
The MSIs do not work. They simply prompt to repair or uninstall and then fail. How would you install this for the first time?

**Reply by tauheedul** [reply]
Hi @mdkimzey943, it needs the OpenXMLSDKV25.msi installing first and then the OpenXMLSDKToolV25.msi

**Comment by mdkimzey943** [comment]
Is there any documentation on how to build the tool from the source?
